### PR TITLE
cmake: Cleanup mimalloc and robin_hood code

### PR DESCRIPTION
**Pull request by juan-lunarg**
Use PUBLIC to ensure dependencies are propogated properly.

Libraries and targets following PUBLIC are linked to, and are made part of the link interface.

This nicely shares the code dependency so additional libraries only have to link against VkLayer_utils.

### CODE REVIEW

File: layers/CMakeLists.txt [hunk 1]
```diff
@@ -81,6 +81,18 @@ if (USE_ROBIN_HOOD_HASHING)
...
 
+# Using mimalloc on non-Windows OSes currently results in unit test instability with some
+# OS version / driver combinations. On 32-bit systems, using mimalloc cause an increase in
+# the amount of virtual address space needed, which can also cause stability problems.
+if (WIN32 AND CMAKE_SIZEOF_VOID_P EQUAL 8)
+   find_package(mimalloc CONFIG)
+   option(USE_MIMALLOC "Use mimalloc, a fast malloc/free replacement library" ${mimalloc_FOUND})
+   if (USE_MIMALLOC)
+      target_compile_definitions(VkLayer_utils PUBLIC USE_MIMALLOC)
+      target_link_libraries(VkLayer_utils PUBLIC mimalloc-static)
```
Comment (juan-lunarg) [inline layers/CMakeLists.txt:92]: The important part is the PUBLIC keyword here:
https://leimao.github.io/blog/CMake-Public-Private-Interface/

TLDR: By linking against VkLayer_utils you also link against mimalloc and robinhood via the propagated dependency.

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM

File: layers/CMakeLists.txt (hunk 1 above)
Comment (arno-lunarg) [reply layers/CMakeLists.txt:92]: Thank you for the explanation!
